dockstore-client/src/test/java/io/dockstore/client/cli/LaunchNoInternetTestIT.java

```diff
@@ -0,0 +1,304 @@
+            add(configPath.getAbsolutePath());
+        }};
+        Client.main(args.toArray(new String[0]));
+    }
+
+    /**
+     * Docker image directory specified but has no files in there
+     */
+    @Test
+    public void directorySpecifiedButNoImages() {
+        checkFailed();
+        exit.checkAssertionAfterwards(()-> assertTrue(systemOutRule.getLog().contains("There are no files in the docker image directory")));
+
+        Path emptyTestDirectory = createEmptyTestDirectory();
+        if (emptyTestDirectory == null) {
+            Assert.fail("Could not create empty temp directory");
+        }
+        String toWrite = "docker-images = " + emptyTestDirectory.toAbsolutePath();
+        File configPath = createTempFile(toWrite);
+        if (configPath == null) {
+            Assert.fail("Could not create temp config file");
+        }
+        ArrayList<String> args = new ArrayList<String>() {{
+            add("tool");
+            add("launch");
+            add("--local-entry");
+            add(DESCRIPTOR_FILE.getAbsolutePath());
+            add("--yaml");
+            add(YAML_FILE.getAbsolutePath());
+            add("--config");
+            add(configPath.getAbsolutePath());
+        }};
+        Client.main(args.toArray(new String[0]));
+    }
+
+    /**
+     * Everything correctly configured with CWL tool
+     */
+    @Test
+    public void correctCWL() {
+        File descriptorFile = new File(ResourceHelpers.resourceFilePath("nonexistent_image/CWL/nonexistent_image.cwl"));
+        String toWrite = "docker-images = " + docker_images;
+        File configPath = createTempFile(toWrite);
+        if (configPath == null) {
+            Assert.fail("Could create temp config file");
+        }
+        ArrayList<String> args = new ArrayList<String>() {{
+            add("tool");
+            add("launch");
+            add("--local-entry");
+            add(descriptorFile.getAbsolutePath());
+            add("--yaml");
+            add(YAML_FILE.getAbsolutePath());
+            add("--config");
+            add(configPath.getAbsolutePath());
+        }};
+        Client.main(args.toArray(new String[0]));
+        Assert.assertTrue("Final process status was not success", systemOutRule.getLog().contains("Final process status is success"));
+    }
+
+    /**
+     * Everything correctly configured with NFL workflow
+     */
+    @Test
+    public void correctNFL() throws IOException {
+        copyNFLFiles();
+        File descriptorFile = new File(ResourceHelpers.resourceFilePath("nonexistent_image/NFL/nextflow.config"));
+        File jsonFile = new File(ResourceHelpers.resourceFilePath("nextflow_rnatoy/test.json"));
+        String toWrite = "docker-images = " + docker_images;
+        File configPath = createTempFile(toWrite);
+        if (configPath == null) {
+            Assert.fail("Could create temp config file");
+        }
+        ArrayList<String> args = new ArrayList<String>() {{
+            add("workflow");
```

Comment:
We can launch Nextflow workflows this way? @agduncan94 
Missed this (might be getting late)